pkg/controller/scyllanodeconfig/sync.go

```diff
@@ -0,0 +1,291 @@
+// Copyright (C) 2021 ScyllaDB
+
+package scyllanodeconfig
+
+import (
+	"context"
+	"fmt"
+
+	scyllav1alpha1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1alpha1"
+	"github.com/scylladb/scylla-operator/pkg/controller/scyllanodeconfig/resource"
+	"github.com/scylladb/scylla-operator/pkg/controllertools"
+	"github.com/scylladb/scylla-operator/pkg/naming"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/client-go/tools/cache"
+)
+
+func (sncc *Controller) sync(ctx context.Context, key string) error {
+	_, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		return fmt.Errorf("split meta namespace cache key: %w", err)
+	}
+
+	snc, err := sncc.scyllaNodeConfigLister.Get(name)
+	if err != nil {
+		if apierrors.IsNotFound(err) && name == resource.DefaultScyllaNodeConfig().Name {
+			snc, err = sncc.scyllaClient.ScyllaNodeConfigs().Create(ctx, resource.DefaultScyllaNodeConfig(), metav1.CreateOptions{})
+			if err != nil {
+				return fmt.Errorf("create default ScyllaNodeConfig: %w", err)
+			}
+			return nil
+		} else {
+			return fmt.Errorf("get ScyllaNodeConfig %q: %w", name, err)
+		}
+	}
+
+	scyllaPods, err := sncc.podLister.Pods(corev1.NamespaceAll).List(naming.ScyllaSelector())
```

Comment:
same here

Reply:
? I can't set fieldselector in podLister